src/widget/wglwidgetqglwidget.cpp

```diff
@@ -0,0 +1,34 @@
+#include <QWindow>
+
+#include "waveform/sharedglcontext.h"
+#include "widget/wglwidget.h"
+
+WGLWidget::WGLWidget(QWidget* parent)
+        : QGLWidget(parent, SharedGLContext::getWidget()) {
+    setAttribute(Qt::WA_NoSystemBackground);
+    setAttribute(Qt::WA_OpaquePaintEvent);
+    setAutoFillBackground(false);
+    setAutoBufferSwap(false);
+    // Not interested in repaint or update calls, as we draw from the vsync thread
+    setUpdatesEnabled(false);
+}
+
+bool WGLWidget::isContextValid() const {
+    return context()->isValid();
```

Comment:
is context always not null? Please add a comment if so.

Reply:
Rather than a comment, in such cases I prefer an assert to express that "as a developer I never expect that to be null". Would that work for you too?

Reply:
Yes, of cause. However that will only make a crash more descriptive in a developer build. My original question was more, if there is a theoretical condition/exceptional code that this can become null?

Reply:
from what I understand from the QGLWidget documentation, it will never be null